packages/core/parcel-bundler/src/SourceMap.js

```diff
@@ -38,7 +38,13 @@ class SourceMap {
   }
 
   async addMap(map, lineOffset = 0, columnOffset = 0) {
+    if (map.toJSON) {
+      // SourceMapGenerator
+      map = map.toJSON();
+    }
+
     if (!(map instanceof SourceMap) && map.version) {
```

Comment:
Probably should be changed to

```Javascript
if ((!(map instanceof SourceMap) && map.version) || typeof map === 'string') {
```

To be able to handle stringified maps (as mentioned in one of my later comments), I probably forgot to add this to this function (it's already part of getConsumer)